VirtoCommerce.PricingModule.Data/Search/ProductPriceDocumentChangesProvider.cs

```diff
@@ -90,15 +90,30 @@ protected virtual async Task<GenericSearchResult<IndexDocumentChange>> InnerGetC
             using (var platformRepository = _platformRepositoryFactory())
             using (var repository = _repositoryFactory())
             {
-                var operationLogChangesQuery = platformRepository.OperationLogs.Where(x => x.ObjectType == _changeLogObjectType && (startDate == null || x.ModifiedDate >= startDate) && (endDate == null || x.ModifiedDate < endDate))
-                                                                 .OrderBy(x => x.ModifiedDate);
-                result.TotalCount = operationLogChangesQuery.Count();
+                // NOTE: we intentionally ignore pagination here and read all changes that happened during given time interval.
+                //       This allows to find IDs of changed products more efficiently and to avoid redundant product reindexing.
+                //       Only priceIds are retrieved from the database, so the memory consumption shouldn't be large.
+                var priceChangeLogEntries = platformRepository.OperationLogs
```

Comment:
However, I would like to do these operations with paging:
- loading all changed price ids in memory
- loading changed prices based on loaded  prices ids

Reply:
If I got it correctly, it should be done now (see the `GetProductIdsForChangedPrices()` method).